### DISCUSSION TITLE
General Contact Question

### DISCUSSION THREAD

**Discussion by makeclean** [idaholab/moose#25630] [Q&A General]
Hi, quick one I think, with regards to contact, when surfaces are considered being in contact, would that impact all variables? I.e. would the solution of different physics/variables also be impacted other than those for displacements? For example, If I were doing a mechanical-electrical coupling, would the contact that is done mechanically allow 'current' to flow? Or would that have to be enabled elsewhere?

**Comment by GiudGiud** [comment]
Hello

No unless you define the mortar contact for these variables too. Which should be possible in quite the same way? Depends what the numerical scheme looks like

Guillaume

**Reply by lindsayad** [reply]
Right, if you want some physics across a mortar interface, whether in contact or not, you will have to code that physics in a mortar constraint object

**Reply by makeclean** [reply]
Ok, for now if I didnt, could I do that all natively with MOOSE input syntax? I have a niche use case where, I would like to apply a Neumann boundary condition on the nodes that are contact, that I presume would be possible writing my own App, but would that possible with a 'no code' solution?

**Reply by GiudGiud** [reply]
Is this an equal gradient type of situation? We have this constraint
https://mooseframework.inl.gov/moose/source/constraints/EqualGradientConstraint.html

or is there a jump due to different diffusivities or some interface resistance?

**Reply by makeclean** [reply]
Yes it would be equal gradient, basically its to mimic the source of friction due to the two entities being in contact, whilst one of them is rotating.

**Reply by GiudGiud** [reply]
The mechanical friction has its own sets of constraints iirc. 
@recuero will know. 
So the problem is mechanical frictional contact + electrical contact?

**Reply by recuero** [reply]
Equal gradient and contact mechanics aren't necessarily analogous. For the same contact pressure, you could get very different gradients in the two bodies in contact, which would depend on the material models.

**Reply by recuero** [reply]
Yes, Coulomb friction has its own set of constraints.

**Comment by makeclean** [comment]
just mechanical friction + thermal, I was using electrical as an analogue, but I would be interested in electrical-mechanical contact in another downstream application. I did search the documentation and objects for 'friction' and stuff related to temperature, as it really the DT due to heat from friction im intersted in

**Reply by recuero** [reply]
That's seems doable. You can compute the energy dissipated due to friction (maybe via an auxiliary kernel) and then use it as a heat source term?
